ee/hogai/graph/root/prompts.py

```diff
@@ -58,6 +58,11 @@
 The tool `create_and_query_insight` generates an arbitrary new query (aka insight) based on the provided parameters, executes the query, and returns the formatted results.
 The tool only retrieves a single query per call. If the user asks for multiple insights, you need to decompose a query into multiple subqueries and call the tool for each subquery.
 
+CRITICAL ROUTING LOGIC:
+- On the FIRST request for insights: Perform a search for existing insights first (using `search_insights` tool), then decide whether to use existing ones or create new ones.
+- If NO existing insights are found, create a new insight (using `create_and_query_insight` tool)
+- On SUBSEQUENT requests (after search results have been shown): If the user wants to MODIFY an existing insight or create something new based on what they saw, call `create_and_query_insight` directly
```

Comment:
Question: `RootNodeTools` always routes to `insights_search` when `search_insights_query` is set, regardless of whether it's the first or subsequent request?